src/netconfig.rs

```diff
@@ -71,8 +66,60 @@ fn handle_netconfig_values() -> Result<Option<Netconfig>, anyhow::Error> {
     Ok(Some(netconfig))
 }
 
+pub fn apply_dns_policy(
+    netconfig: &Netconfig,
+    nm_state: &mut NetworkState,
+) -> Result<(), anyhow::Error> {
+    // Start at 10 because 0 is special global default in NM
+    // and increase by 10 to give room for future changes.
+    let mut i: i32 = 10;
+    for policy in &netconfig.dns_policy {
+        match policy.as_str() {
+            "" => continue,
+            "STATIC" => {
+                let mut loopback = match nm_state.get_connection("lo") {
+                    Some(lo) => lo.clone(),
+                    None => anyhow::bail!("Failed to get loopback connection"),
+                };
+                loopback.ip_config.dns_priority4 = Some(i);
+                loopback.ip_config.dns_priority6 = Some(i);
+                nm_state.update_connection(loopback)?;
+            }
+            _ => {
+                let glob = Glob::new(policy)?.compile_matcher();
+                for con in nm_state
+                    .connections
+                    .iter()
+                    .map(|c| c.id.clone())
+                    .filter(|c_id| glob.is_match(c_id))
+                    .collect::<Vec<String>>()
+                {
+                    let mut con = match nm_state.get_connection(&con) {
+                        Some(con) => con.clone(),
+                        None => anyhow::bail!("Failed to get connection from state: {}", con),
+                    };
+                    // If con was already matched its priority should not be overwritten
+                    if con.ip_config.dns_priority4.is_some()
+                        && con.ip_config.dns_priority6.is_some()
+                    {
+                        continue;
+                    }
+                    con.ip_config.dns_priority4 = Some(i);
+                    con.ip_config.dns_priority6 = Some(i);
+                    nm_state.update_connection(con)?;
```

Comment:
```suggestion
                    let Some(con) = nm_state.get_connection_mut(&con) else {
                        anyhow::bail!("Failed to get connection from state: {}", con);
                    };
                    // If con was already matched its priority should not be overwritten
                    if con.ip_config.dns_priority4.is_some()
                        && con.ip_config.dns_priority6.is_some()
                    {
                        continue;
                    }
                    con.ip_config.dns_priority4 = Some(i);
                    con.ip_config.dns_priority6 = Some(i);
```

Reply:
hm I think we can simplify the full loop:
```
                let glob = Glob::new(policy)?.compile_matcher();
                for con in nm_state
                    .connections
                    .iter_mut()
                    .filter(|c| glob.is_match(&c.id))
                    .collect::<Vec<&mut Connection>>()
                {
                    // If con was already matched its priority should not be overwritten
                    if con.ip_config.dns_priority4.is_some()
                        && con.ip_config.dns_priority6.is_some()
                    {
                        continue;
                    }
                    con.ip_config.dns_priority4 = Some(i);
                    con.ip_config.dns_priority6 = Some(i);
                }
```

Reply:
Yeah you are right, I thought I tested that the `update_connection` call is necessary, but it seems I was wrong.